### PR TITLE
Fix NPE issue

### PR DESCRIPTION
**Pull request by nilasini**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtImpl.java [hunk 1]
```diff
@@ -86,8 +88,15 @@ public String addTenant(Tenant tenant) throws TenantMgtException {
...
+
             if (INVITE_VIA_EMAIL.equalsIgnoreCase(tenant.getProvisioningMethod())) {
-                tenant.getClaimsMap().put(TENANT_ADMIN_ASK_PASSWORD_CLAIM, "true");
+                if (tenant.getClaimsMap() != null) {
+                    tenant.getClaimsMap().put(TENANT_ADMIN_ASK_PASSWORD_CLAIM, "true");
+                } else {
+                    Map<String, String> claimsMap = new HashMap<>();
+                    claimsMap.put(TENANT_ADMIN_ASK_PASSWORD_CLAIM, "true");
+                    tenant.setClaimsMap(claimsMap);
+                }
```
Comment (thanujalk) [inline components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtImpl.java:99]: if (tenant.getClaimsMap() == null) {
    tenant.setClaimsMap(new HashMap<>());
}
tenant.getClaimsMap().put(TENANT_ADMIN_ASK_PASSWORD_CLAIM, "true");

File: components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtImpl.java [hunk 2]
```diff
@@ -86,7 +87,11 @@ public String addTenant(Tenant tenant) throws TenantMgtException {
 
             // For the super tenant tenant creation, tenants are always activated as they are created.
             TenantMgtUtil.activateTenantInitially(tenantInfoBean, tenantId);
+
             if (INVITE_VIA_EMAIL.equalsIgnoreCase(tenant.getProvisioningMethod())) {
+                if (tenant.getClaimsMap() == null) {
```
Comment (IsuraD) [inline components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtImpl.java:92]: Shall we fix this at tenant object level? It should not return null in the tenant.getClaimsMap() API

Comment (nilasini) [reply components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtImpl.java:92]: ok fixed with https://github.com/wso2/carbon-kernel/pull/2683

Comment (nilasini) [reply components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtImpl.java:92]: Fixed with https://github.com/wso2/carbon-kernel/pull/2683

**Comment by nilasini** [comment]
closing since fixed with https://github.com/wso2/carbon-kernel/pull/2683

**Pull request closed without merging**